ikarus/finiteelements/mechanics/materials/strainconversions.hh

```diff
@@ -125,14 +125,14 @@ decltype(auto) createRightCauchyGreen(const Eigen::MatrixBase<Derived>& eMB) {
  * \tparam from Type of the source strain tag.
  * \tparam to Type of the target strain tag.
  * \tparam Derived Type of the Eigen matrix.
- * \param eRaw Eigen matrix representing the input strain.
+ * \param eRaw Eigen matrix representing the input strain (can be in Voigt notation).
  * \return The transformed strain matrix.
  */
 template <StrainTags from, StrainTags to, typename Derived>
 decltype(auto) transformStrain(const Eigen::MatrixBase<Derived>& eRaw) {
   static_assert((from == to) or (from != StrainTags::linear and to != StrainTags::linear),
                 "No useful transformation available for linear strains.");
-  decltype(auto) e = Impl::mayTransformFromVoigt(eRaw, true);
+  decltype(auto) e = Impl::mayTransformFromVoigt(eRaw.eval(), true);
```

Comment:
Why is the eval needed here? Did you see some dangling references?

Reply:
I can't quite remember, but I had a bug under certain situations, where the e was always nan, and this fixed that